evalml/pipelines/components/transformers/feature_selection/rf_classifier_feature_selector.py

```diff
@@ -29,11 +29,11 @@ class RFClassifierSelectFromModel(FeatureSelector):
     name = "RF Classifier Select From Model"
     hyperparameter_ranges = {
         "percent_features": Real(0.01, 1),
-        "threshold": ["mean", -np.inf],
+        "threshold": ["mean"],
```

Comment:
This PR proposes using the `mean` as the threshold but experiments show that `median` performs similarly as well. `median` will select exactly half of the features and `mean` depending on the distribution. Happy to discuss which one to choose but I chose `mean` in the... meantime..

Reply:
On second thought: using `median` might be the move as the `mean` of feature importances will bound to be dragged down by low signal features (which are inevitable). However, performance results show similar model quality between `median` and `mean` but `median` having slightly longer fit times due to `median` selecting more features.

Reply:
Why not allow both 'mean' and 'median' to be in the hyperparameter ranges? We can default to one, but allowing both in the ranges seems to be ideal for our automlsearch algo

Reply:
I like that suggestion. Right now the DefaultAlgorithm only runs one feature selector, so I'm down to set median as the default. Yesterday we discussed letting the algorithm tune the parameters of the selector in which case broadening the threshold search space (like parametrizing it as a quantile of the observed feature importance distribution) will be in play.

Reply:
@bchen1116 @freddyaboulton 

sounds good, I'll add them both as hyperparameter ranges! My main concern is what Freddy brought up about the default parameter and having only 1 FS batch but I'm still on the fence on chosing `mean` or `median`. Either way it'll be a quick fix so I'm not too worried!